pkg/util/helper/binding.go

```diff
@@ -162,54 +160,38 @@ func FetchWorkload(dynamicClient dynamic.Interface, restMapper meta.RESTMapper,
 }
 
 // EnsureWork ensure Work to be created or updated.
-//nolint:gocyclo
-// Note: ignore the cyclomatic complexity issue to get gocyclo on board. Tracked by: https://github.com/karmada-io/karmada/issues/460
-func EnsureWork(c client.Client, workload *unstructured.Unstructured, clusterNames []string, scheduleResult []workv1alpha1.TargetCluster,
-	overrideManager overridemanager.OverrideManager, binding metav1.Object, scope apiextensionsv1.ResourceScope) error {
-	var desireReplicaInfos map[string]int64
-	var referenceRSP *v1alpha1.ReplicaSchedulingPolicy
-	var err error
-	hasScheduledReplica := HasScheduledReplica(scheduleResult)
-	if hasScheduledReplica {
-		desireReplicaInfos = transScheduleResultToMap(scheduleResult)
-	} else {
-		referenceRSP, desireReplicaInfos, err = calculateReplicasIfNeeded(c, workload, clusterNames)
-		if err != nil {
-			klog.Errorf("Failed to get ReplicaSchedulingPolicy for %s/%s/%s, err is: %v", workload.GetKind(), workload.GetNamespace(), workload.GetName(), err)
-			return err
-		}
+func EnsureWork(c client.Client, workload *unstructured.Unstructured, overrideManager overridemanager.OverrideManager, binding metav1.Object, scope apiextensionsv1.ResourceScope) error {
+	var targetClusters []workv1alpha1.TargetCluster
+	switch scope {
+	case apiextensionsv1.NamespaceScoped:
+		bindingObj := binding.(*workv1alpha1.ResourceBinding)
+		targetClusters = bindingObj.Spec.Clusters
+	case apiextensionsv1.ClusterScoped:
+		bindingObj := binding.(*workv1alpha1.ClusterResourceBinding)
+		targetClusters = bindingObj.Spec.Clusters
 	}
 
-	var workLabel = make(map[string]string)
+	hasScheduledReplica, referenceRSP, desireReplicaInfos, err := getRSPAndReplicaInfos(c, workload, targetClusters)
+	if err != nil {
+		return err
+	}
 
+	clusterNames := GetBindingClusterNames(targetClusters)
```

Comment:
You can just range the `targetClusters` like this:
```golang
for _, targetCluster := range targetClusters{
```

Reply:
I have modified it.